ingesters/MSGraphIngester/main.go

```diff
@@ -0,0 +1,451 @@
+)
+
+func init() {
+	flag.Parse()
+	if *ver {
+		version.PrintVersion(os.Stdout)
+		ingest.PrintVersion(os.Stdout)
+		os.Exit(0)
+	}
+	lg = log.New(os.Stderr) // DO NOT close this, it will prevent backtraces from firing
+	if *stderrOverride != `` {
+		if oldstderr, err := syscall.Dup(int(os.Stderr.Fd())); err != nil {
+			lg.Fatal("Failed to dup stderr: %v\n", err)
+		} else {
+			lg.AddWriter(os.NewFile(uintptr(oldstderr), "oldstderr"))
+		}
+
+		fp := path.Join(`/dev/shm/`, *stderrOverride)
+		fout, err := os.Create(fp)
+		if err != nil {
+			fmt.Fprintf(os.Stderr, "Failed to create %s: %v\n", fp, err)
+		} else {
+			version.PrintVersion(fout)
+			ingest.PrintVersion(fout)
+			//file created, dup it
+			if err := syscall.Dup2(int(fout.Fd()), int(os.Stderr.Fd())); err != nil {
+				fmt.Fprintf(os.Stderr, "Failed to dup2 stderr: %v\n", err)
+				fout.Close()
+			}
+		}
+	}
+}
+
+type event struct {
+	Id string
+}
+
+func main() {
+	cfg, err := GetConfig(*configLoc)
+	if err != nil {
+		lg.Fatal("Failed to get configuration: %v", err)
+	}
+	if len(cfg.Global.Log_File) > 0 {
+		fout, err := os.OpenFile(cfg.Global.Log_File, os.O_APPEND|os.O_CREATE|os.O_WRONLY, 0640)
+		if err != nil {
+			lg.FatalCode(0, "Failed to open log file %s: %v", cfg.Global.Log_File, err)
+		}
+		if err = lg.AddWriter(fout); err != nil {
+			lg.Fatal("Failed to add a writer: %v", err)
+		}
+		if len(cfg.Global.Log_Level) > 0 {
+			if err = lg.SetLevelString(cfg.Global.Log_Level); err != nil {
+				lg.FatalCode(0, "Invalid Log Level \"%s\": %v", cfg.Global.Log_Level, err)
+			}
+		}
+	}
+
+	tags, err := cfg.Tags()
+	if err != nil {
+		lg.Fatal("Failed to get tags from configuration: %v", err)
+	}
+	conns, err := cfg.Targets()
+	if err != nil {
+		lg.Fatal("Failed to get backend targets from configuration: %s", err)
+	}
+	debugout("Handling %d tags over %d targets\n", len(tags), len(conns))
+
+	//fire up the ingesters
+	id, ok := cfg.Global.IngesterUUID()
+	if !ok {
+		lg.FatalCode(0, "Couldn't read ingester UUID\n")
+	}
+	ingestConfig := ingest.UniformMuxerConfig{
+		IngestStreamConfig: cfg.Global.IngestStreamConfig,
+		Destinations:       conns,
+		Tags:               tags,
+		Auth:               cfg.Secret(),
+		LogLevel:           cfg.LogLevel(),
+		Logger:             lg,
+		IngesterName:       "Microsoft Graph",
+		IngesterVersion:    version.GetVersion(),
+		IngesterUUID:       id.String(),
+		CacheDepth:         cfg.Global.Cache_Depth,
+		CachePath:          cfg.Global.Ingest_Cache_Path,
+		CacheSize:          cfg.Global.Max_Ingest_Cache,
+		CacheMode:          cfg.Global.Cache_Mode,
+	}
+	igst, err := ingest.NewUniformMuxer(ingestConfig)
+	if err != nil {
+		lg.Fatal("Failed build our ingest system: %v", err)
+	}
+	defer igst.Close()
+	debugout("Starting ingester muxer\n")
+	if err := igst.Start(); err != nil {
+		lg.FatalCode(0, "Failed start our ingest system: %v", err)
+		return
+	}
+
+	debugout("Waiting for connections to indexers ... ")
```

Comment:
missing newline?

Reply:
The intent is that it'll show "Wait for connections to indexers ... Successfully connected to ingesters"

Reply:
ok